pkg/autogen/autogen.go

```diff
@@ -98,6 +100,53 @@ func CanAutoGen(spec *kyverno.Spec, log logr.Logger) (applyAutoGen bool, control
 	return true, engine.PodControllers
 }
 
+// GetSupportedControllers returns the supported autogen controllers for a given spec.
+func GetSupportedControllers(spec *kyverno.Spec, log logr.Logger) []string {
+	apply, controllers := CanAutoGen(spec, log)
+	if !apply || controllers == "none" {
+		return nil
+	}
+	return strings.Split(controllers, ",")
+}
+
+// GetRequestedControllers returns the requested autogen controllers based on object annotations.
+func GetRequestedControllers(meta metav1.ObjectMeta) []string {
+	annotations := meta.GetAnnotations()
+	if annotations == nil {
+		return nil
+	}
+	controllers, ok := annotations[engine.PodControllersAnnotation]
```

Comment:
I think it's not necessary for now. We discussed it on slack and I'm not sure we will need a v2 annotation TBH.

Reply:
So we will use the same annotation for both:

- generate additional auto-gen rules and add them to `spec.rules`
- update `status.autogen` field

Obviously the first will be deprecated in 1.7.0 and removed in 1.9.0. 

I'm trying to understand how to use them internally. With this change, the policy will have both blocks indicating autogen rules, I would assume we will use `status.autogen` to process the resource inside the engine. This means autogen rules in `spec.rules` will not be used at all. Is this the case?

It's a bit unclear to me what are next steps in order to fully support autogen via policy status. Can you help clarify?

Reply:
What would a `v2` annotation bring over what we already have ?
The current annotation holds enough semantics for us to do our work IMHO, and introducing a `v2` could be quite confusing.

I feel like we're almost done, next steps would be:
- Stop writing to the annotation, only use it for reads
- Compute autogen rules and store the full set of rules in the status in config manager reconciliation loop (give precedence to existing rules if any ?)
- Fetch rules from the status rather than from the spec (i already added a `GetRules` method so it should be easy to plumb this logic here)

Reply:
How does that sound ?

Reply:
Isn’t autogen applied by default when the annotation is not present ?

Reply:
No, currently we mutate the annotation if it's empty.

Reply:
At first sight it would have the same side effects to apply all controllers when the annotation is absent.